src/Data/Nat/DivMod.agda

```diff
@@ -115,6 +115,26 @@ m<[1+n%d]⇒m≤[n%d] {m} n (suc d-1) = k<1+a[modₕ]n⇒k≤a[modₕ]n 0 m n d-
   (m % d +  n % d + (n / d) * d)  % d ≡⟨ [m+kn]%n≡m%n (m % d + n % d) (n / d) d-1 ⟩
   (m % d +  n % d)                % d ∎
 
+%-distribˡ-* : ∀ m n d {≢0} → ((m * n) % d) {≢0} ≡ (((m % d) {≢0} * (n % d) {≢0}) % d) {≢0}
+%-distribˡ-* m n d@(suc d-1) = begin-equality
+  (m * n)                                             % d ≡⟨ cong (λ h → (h * n) % d) (m≡m%n+[m/n]*n m d-1) ⟩
```

Comment:
The first 9 steps here all seem to be manipulations under the outer `% d` - why not pull that out as a lemma?

Reply:
In my opinion I would keep as is, but use the new `Data.Nat.Tactic.RingSolver` to automatically generate the proof for step 4 to step 11.

Reply:
I dont know how to use tactics in agda and I couldn't find introduction to it.